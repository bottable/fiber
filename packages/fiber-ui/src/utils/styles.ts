import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
