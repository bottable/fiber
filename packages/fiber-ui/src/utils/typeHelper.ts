export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
