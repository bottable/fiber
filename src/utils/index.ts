export * from './typeHelper'
export * from './mergeProps'
