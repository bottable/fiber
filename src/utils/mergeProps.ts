import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
