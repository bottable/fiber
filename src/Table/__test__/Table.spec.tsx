import { Table } from '..'

import React from 'react'
import { render } from 'test/utils'

describe('Table', () => {
  it('renders correctly', () => {
    const table = render(<Table />)

    expect(table).toMatchSnapshot()
  })
})
