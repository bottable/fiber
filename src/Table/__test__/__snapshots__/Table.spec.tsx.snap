// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
<div
  className="sc-fzoYkl BkXxB"
>
  <div
    className="sc-fzpkJw gEiHfE"
  >
    <table
      className="sc-fznzOf kmZbJz"
    >
      <thead
        className="sc-fznMnq eOdCaO"
      >
        <tr>
          <th
            className="sc-fzomME dYKiYg"
          >
            Name
          </th>
          <th
            className="sc-fzomME dYKiYg"
          >
            Age
          </th>
          <th
            className="sc-fzomME dYKiYg"
          >
            Address
          </th>
        </tr>
      </thead>
      <tbody
        className="sc-fzoant cATlvS"
        onChange={[Function]}
      >
        <tr
          className="sc-fzqAbL ddqibk"
        >
          <td
            className="sc-fzqMAW kKFpfm"
          >
            <a>
              John Brown
            </a>
          </td>
          <td
            className="sc-fzqMAW kKFpfm"
          >
            32
          </td>
          <td
            className="sc-fzqMAW kKFpfm"
          >
            New York No. 1 Lake Park
          </td>
        </tr>
        <tr
          className="sc-fzqAbL ddqibk"
        >
          <td
            className="sc-fzqMAW kKFpfm"
          >
            <a>
              Jim Green
            </a>
          </td>
          <td
            className="sc-fzqMAW kKFpfm"
          >
            42
          </td>
          <td
            className="sc-fzqMAW kKFpfm"
          >
            London No. 1 Lake Park
          </td>
        </tr>
        <tr
          className="sc-fzqAbL ddqibk"
        >
          <td
            className="sc-fzqMAW kKFpfm"
          >
            <a>
              Joe Black
            </a>
          </td>
          <td
            className="sc-fzqMAW kKFpfm"
          >
            32
          </td>
          <td
            className="sc-fzqMAW kKFpfm"
          >
            Sidney No. 1 Lake Park
          </td>
        </tr>
        <tr
          className="sc-fzqAbL ddqibk"
        >
          <td
            className="sc-fzqMAW kKFpfm"
          >
            <a>
              Disabled User
            </a>
          </td>
          <td
            className="sc-fzqMAW kKFpfm"
          >
            99
          </td>
          <td
            className="sc-fzqMAW kKFpfm"
          >
            Sidney No. 1 Lake Park
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <ul
    className="sc-fzolEj hGURrG sc-fzpkqZ daiBEg"
  >
    <li
      className="sc-fzqBkg eDmeRO"
    >
      <button
        className="sc-fzpans sc-fzokOt eBuWRk sc-fzqPZZ hruXgM"
        disabled={true}
        onClick={[Function]}
        onMouseDown={[Function]}
        selected={false}
      >
        <span
          className="sc-fzplWN dsgbOd"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </span>
      </button>
    </li>
    <li
      className="sc-fzqBkg eDmeRO"
      onClick={[Function]}
    >
      <button
        className="sc-fzpans sc-fzokOt czwoiC sc-fzqPZZ eVTpHQ"
        onClick={[Function]}
        onMouseDown={[Function]}
        selected={true}
      >
        <span>
          1
        </span>
      </button>
    </li>
    <li
      className="sc-fzqBkg eDmeRO"
    >
      <button
        className="sc-fzpans sc-fzokOt eBuWRk sc-fzqPZZ hruXgM"
        disabled={true}
        onClick={[Function]}
        onMouseDown={[Function]}
        selected={false}
      >
        <span
          className="sc-fzplWN dsgbOd"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </span>
      </button>
    </li>
  </ul>
</div>
`;
